pkg/kstatus/polling/engine/engine.go

```diff
@@ -204,17 +214,25 @@ func (r *statusPollerRunner) syncAndPoll() error {
 	// Poll all resources and compute status. If the polling of resources has completed (based
 	// on information from the StatusAggregator and the value of pollUntilCancelled), we send
 	// a CompletedEvent and return.
-	r.pollStatusForAllResources()
-	return nil
+	return r.pollStatusForAllResources()
 }
 
 // pollStatusForAllResources iterates over all the resources in the set and delegates
 // to the appropriate engine to compute the status.
-func (r *statusPollerRunner) pollStatusForAllResources() {
+func (r *statusPollerRunner) pollStatusForAllResources() error {
 	for _, id := range r.identifiers {
+		// Check if the context has been cancelled on every iteration.
+		select {
+		case <-r.ctx.Done():
+			return r.ctx.Err()
```

Comment:
Why not just return nil here? Then Run doesn't need to catch it and ignore it.

Reply:
The idea was that since context errors from the `ReadStatus` are passed back, it would be consistent if we returned them here too. That way we can catch them all in `Run`.